In [1]:
import csv
f=open('guns.csv','r')
read=csv.reader(f)
data=list(read)
print(data[0:5])

[['', 'year', 'month', 'intent', 'police', 'sex', 'age', 'race', 'hispanic', 'place', 'education'], ['1', '2012', '01', 'Suicide', '0', 'M', '34', 'Asian/Pacific Islander', '100', 'Home', '4'], ['2', '2012', '01', 'Suicide', '0', 'F', '21', 'White', '100', 'Street', '3'], ['3', '2012', '01', 'Suicide', '0', 'M', '60', 'White', '100', 'Other specified', '4'], ['4', '2012', '02', 'Suicide', '0', 'M', '64', 'White', '100', 'Home', '4']]


In [2]:
headers=data[0]
data=data[1:len(data)]
print(data[0:5])

[['1', '2012', '01', 'Suicide', '0', 'M', '34', 'Asian/Pacific Islander', '100', 'Home', '4'], ['2', '2012', '01', 'Suicide', '0', 'F', '21', 'White', '100', 'Street', '3'], ['3', '2012', '01', 'Suicide', '0', 'M', '60', 'White', '100', 'Other specified', '4'], ['4', '2012', '02', 'Suicide', '0', 'M', '64', 'White', '100', 'Home', '4'], ['5', '2012', '02', 'Suicide', '0', 'M', '31', 'White', '100', 'Other specified', '2']]


In [3]:
years = [row[1] for row in data]
year_counts={}
for year in years:
    if year in year_counts:
        year_counts[year]+=1
    else:
        year_counts[year]=1
print(year_counts)

{'2012': 33563, '2013': 33636, '2014': 33599}


In [4]:
import datetime
dates = [datetime.datetime(year=int(row[1]), month=int(row[2]), day=1) for row in data]
print(dates[0:5])
date_counts={}
for date in dates:
    if date in date_counts:
        date_counts[date]+=1
    else:
        date_counts[date]=1
print("\n")
print(date_counts)



[datetime.datetime(2012, 1, 1, 0, 0), datetime.datetime(2012, 1, 1, 0, 0), datetime.datetime(2012, 1, 1, 0, 0), datetime.datetime(2012, 2, 1, 0, 0), datetime.datetime(2012, 2, 1, 0, 0)]


{datetime.datetime(2014, 8, 1, 0, 0): 2970, datetime.datetime(2013, 6, 1, 0, 0): 2920, datetime.datetime(2014, 5, 1, 0, 0): 2864, datetime.datetime(2013, 2, 1, 0, 0): 2375, datetime.datetime(2012, 8, 1, 0, 0): 2954, datetime.datetime(2012, 7, 1, 0, 0): 3026, datetime.datetime(2013, 10, 1, 0, 0): 2808, datetime.datetime(2012, 3, 1, 0, 0): 2743, datetime.datetime(2014, 1, 1, 0, 0): 2651, datetime.datetime(2012, 11, 1, 0, 0): 2729, datetime.datetime(2014, 12, 1, 0, 0): 2857, datetime.datetime(2012, 1, 1, 0, 0): 2758, datetime.datetime(2014, 6, 1, 0, 0): 2931, datetime.datetime(2012, 6, 1, 0, 0): 2826, datetime.datetime(2012, 2, 1, 0, 0): 2357, datetime.datetime(2013, 3, 1, 0, 0): 2862, datetime.datetime(2014, 4, 1, 0, 0): 2862, datetime.datetime(2014, 3, 1, 0, 0): 2684, datetime.datetime(2014, 11, 1, 0, 

In [5]:
sex = [row[5] for row in data]
race = [row[7] for row in data]
def column_count(list):
    counts={}
    for value in list:
        if value in counts:
            counts[value]+=1
        else:
            counts[value]=1
    return counts

sex_counts=column_count(sex)
race_counts=column_count(race)
print(sex_counts,'\n',race_counts)

{'F': 14449, 'M': 86349} 
 {'Black': 23296, 'Hispanic': 9022, 'Asian/Pacific Islander': 1326, 'White': 66237, 'Native American/Native Alaskan': 917}


# Gun Deaths by Gender
Based on this analysis, more men were killed e by gun deaths then were women from 2012 to 2014. Considering the ratio of Males to Females in the US is approximately 1 to 1, this suggests that men are killed at a much higher rate by guns then are females. 

# Gun Deaths By Race
Based on the analysis, the white race has significantly more gun deaths than any other race in the US. They are followed by black, hispanic, native american and asian/pacific islanders respectively. Although interesting, we need data on the population of each race in the US to compare the rate of which each race is killed by gun deaths. 

In [6]:
f1=open('census.csv','r')
read1=csv.reader(f1)
census=list(read1)
print(census)

[['Id', 'Year', 'Id', 'Sex', 'Id', 'Hispanic Origin', 'Id', 'Id2', 'Geography', 'Total', 'Race Alone - White', 'Race Alone - Hispanic', 'Race Alone - Black or African American', 'Race Alone - American Indian and Alaska Native', 'Race Alone - Asian', 'Race Alone - Native Hawaiian and Other Pacific Islander', 'Two or More Races'], ['cen42010', 'April 1, 2010 Census', 'totsex', 'Both Sexes', 'tothisp', 'Total', '0100000US', '', 'United States', '308745538', '197318956', '44618105', '40250635', '3739506', '15159516', '674625', '6984195']]


In [11]:

mapping={'Asian/Pacific Islander':(15834141),'Black':40250635,'Hispanic':44618105,'Native American/Native Alaskan':3739506,'White':197318956
        }
race_per_hundredk={}
for key in race_counts:
    if key in mapping:
        rate_10k=(race_counts[key]/mapping[key])*100000
        race_per_hundredk[key]=rate_10k
print(race_per_hundredk)



{'Black': 57.8773477735196, 'Hispanic': 20.220491210910907, 'Asian/Pacific Islander': 8.374309664161762, 'White': 33.56849303419181, 'Native American/Native Alaskan': 24.521955573811088}


In [14]:
intents=[row[3] for row in data]
races = [row[7] for row in data]
homicide_race_counts = {}
for i,race in enumerate(races):
    if intents[i]== 'Homicide':
        if race in homicide_race_counts:
            homicide_race_counts[race]+=1
        else:
            homicide_race_counts[race]=1
for key in homicide_race_counts:
    rate2100k=(homicide_race_counts[key]/mapping[key])*100000
    homicide_race_counts[key]=rate2100k
print(homicide_race_counts)

{'Black': 48.471284987180944, 'Hispanic': 12.627161104219914, 'Asian/Pacific Islander': 3.530346230970155, 'White': 4.6356417981453335, 'Native American/Native Alaskan': 8.717729026240365}


# Gun Death Rate by Race
## Adding population data into the analysis, I was able to calculate the  the gun death rate per 100,000 people for each race.
There is noticible gun death disparity between different races. Black people are the most likely out of any race in the dataset to be be killed by a gun. White people are next followed by Native americans, Hispanics, and Asians/Pacific Islanders. 

# Gun Murder Rate by Race
## When Limiting Gun Deaths to Murders, Race Disparity Increases
The results: 48 out of every 100,000 black persons is murdered with a gun.  Compared to a white person, black are 12 times more likely to be murdered with a gun. Hispanics also suffer from a high gun murder rate, which is approximately 3 times higher than a white person's. Asian and Pacific Islanders are the least likely to be murdered by a gun.Native Americans and Native Alaskans are twice as likely as white people Their rate per 100,000 people is 3.53. The murder rate by gun for white people is 4.34 out of every 100,000. There is a significant  drop off in the gun death rate for white people compared to the gun murder rate for white people. I'd like to analyze suicide rates next to examine the race disparity. I'd assume that the suicide rate among white people is much higher compared to other races. 

In [15]:
Next steps: Check suicide rates, plot the growth trend from 2012-2014, 

SyntaxError: invalid syntax (<ipython-input-15-4250277de644>, line 1)